hw/misc/ssi_psram.c

```diff
@@ -96,12 +197,25 @@ static void psram_realize(SSIPeripheral *ss, Error **errp)
         error_report("[PSRAM] Invalid size %dMB for the PSRAM", s->size_mbytes);
     }
 
+    /* Set the default MR values for the octal psram (ref: Datasheet APS6408L_OBMx) */
+    s->mr0 = MR0_RD_LT_VARIABLE | MR0_RD_LATENCY_CODE | MR0_DRIVE_STRENGHT_HALF;
+    s->mr1 = MR1_NO_ULP | MR1_VENDOR_ID;
+    s->mr2 = MR2_GOOD_DIE_BIT_PASS | MR2_DEVICE_ID_3_GEN | MR2_DENSITY_MAPPING_64MB;   
+    s->mr3 = MR3_RBX_NOT_SUPPORTED | MR3_OP_VOLTAGE_1V8 | MR3_SRF_FAST_REFRESH;
+    s->mr4 = MR4_WRITE_LATENCY_5 | MR4_FAST_REFRESH | MR4_PASR_64MB;
+    s->mr8 = MR8_RBX_READ_DISABLE | MR8_HYBRID_BURST | MR8_32BYTE_BURST;
+
+    if (s->is_octal) {
+        s->dummy = 3;
+    }
+
     /* Allocate the actual array that will act as a vritual RAM */
     const uint32_t size_bytes = s->size_mbytes * 1024 * 1024;
     memory_region_init_ram(&s->data_mr, OBJECT(s), "psram.memory_region", size_bytes, &error_fatal);
 }
 
 static Property psram_properties[] = {
+    DEFINE_PROP_BOOL("is_octal", SsiPsramState, is_octal, true),
```

Comment:
Would it be more reasonable to keep `false` as the default? For example, esp32 doesn't support octal PSRAM. We can enable it for esp32s3 when the related Kconfig option is set.

Reply:
oups indeed it should be set to false by default.
Is there a way, implemented in Qemu, to change this settings depending on this option ?

Reply:
Not in QEMU, since it isn't aware of ESP-IDF and its options. In IDF's QEMU integration, we can do such things; see e.g. [this function](https://github.com/espressif/esp-idf/blob/b5ac4fbdf9e9fb320bb0a98ee4fbaa18f8566f37/tools/idf_py_actions/qemu_ext.py#L216), we adjust the flash size depending on an sdkconfig option.